### PR TITLE
radiation 1D

### PR DESCRIPTION
**Pull request by UncleRais**
developed 1D thermal radiation boundary condition, added corresponding unit test

### CODE REVIEW

File: tests/nonstationary_thermal_1d/boundary_conditions_1d.cpp [hunk 1]
```diff
@@ -67,6 +67,54 @@ void solve_nonstationary_thermal_1d_problem(const std::shared_ptr<mesh::mesh_1d<
     }
 }
 
+template<std::floating_point T>
+T calc_energy(const heat_equation_solution_1d<T>& solution) {
```
Comment (VIXER13) [inline tests/nonstationary_thermal_1d/boundary_conditions_1d.cpp:71]: Лучше воспользуйся функцией `mesh::integrate`, она располагается в `libraries/mesh/mesh_1d/mesh_1d_utils`

File: tests/nonstationary_thermal_1d/boundary_conditions_1d.cpp [hunk 2]
```diff
@@ -67,6 +67,54 @@ void solve_nonstationary_thermal_1d_problem(const std::shared_ptr<mesh::mesh_1d<
...
+            std::make_unique<Right_bc_type<T>>(right_bc(T(time_layer)))
+        };
+        solver.compute(parameters, boundaries_conditions, EMPTY_FUNCTION);  
+        const auto rp = [&right_part, &time_layer](const T x) { return right_part(time_layer, x); };
+        solver.calc_step(boundaries_conditions, rp);
+        heat_equation_solution_1d<T> solution{mesh, parameters, solver.temperature()};
+        energy.push_back(calc_energy(solution));
+        // The last one is smaller than the previous
+        if(step == time.steps_count) 
+            expect(energy.back() < *(energy.end() - 2));
```
Comment (VIXER13) [inline tests/nonstationary_thermal_1d/boundary_conditions_1d.cpp:112]: Для итераторов лучше использовать функции по типу `std::next` и `std::prev`. Прямые математические операции могут не тот результат дать, потому что итераторы не всегда работают как указатели.

File: tests/nonstationary_thermal_1d/boundary_conditions_1d.cpp [hunk 3]
```diff
@@ -67,6 +67,54 @@ void solve_nonstationary_thermal_1d_problem(const std::shared_ptr<mesh::mesh_1d<
...
+        else if(internal_iters_check)
```
Comment (VIXER13) [inline tests/nonstationary_thermal_1d/boundary_conditions_1d.cpp:113]: Мне кажется тут можно 2 if объединить в один
if (internal_iters_check || step == time.steps_count)
